### ISSUE TITLE
`unhovered` doesn't get a chance to act on loss of focus under certain conditions

### ISSUE THREAD

**Issue by JSearUK** [renpy/renpy#3323]
Dropping the following into a new project should demonstrate the problem:
```py
init:
    default stage = 0
    default button_data = [ ("Hover over the button,\nthen press Shift+A,\nthen click Return.", [NullAction()]),
                            ("Click the button.\nUnhover, rehover.\nBehaves.\nClick Hide.", [Show("CoverScreen")]),
                            ("Click the button,\nthen click Hide.", [Show("CoverScreen", area=(0.125, 0.25, 0.25, 0.5))]),
                            ("Click the button,\nthen click Hide.", [Show("CoverScreen", area=(0.125, 0.125, 0.75, 0.125))]),
                            ("Quit via the menu\nor\nClick elsewhere to restart", [NullAction()])
                          ]


transform scroll():
    ypos 1.0 yanchor 0.0
    linear 5.0 ypos 0.0 yanchor 1.0
    repeat


screen CoverScreen(area=()):
    modal (False if stage == 1 else True) # This
    frame:
        if area:
            area area
        else:
            at truecenter
            xysize (0.75, 0.5)
        background "#FFF4"
        textbutton "Hide":
            action Hide("CoverScreen")


screen DemoButton():
    default hasfocus = False
    button:
        at truecenter
        xysize (0.5, 256)
        background Frame("gui/frame.png")
        foreground "#0008"
        hover_foreground None
        hovered SetScreenVariable("hasfocus", True)
        unhovered SetScreenVariable("hasfocus", False) # This
        action button_data[stage][1]
        side "l c r":
            fixed:
                xysize (256, 256)
                add "gui/window_icon.png"
            fixed:
                at Transform(crop=(0, 0, 1.0, 1.0), crop_relative=True)
                text button_data[stage][0]:
                    align (0.5, 0.5)
                    text_align 0.5
                    if hasfocus:
                        color gui.hover_color
                        at scroll
            fixed:
                xysize (256, 256)
                add "gui/window_icon.png"


label start:
    $ quick_menu = False
    scene black
label loop:
    show screen DemoButton
    "{b}Example 1: Accessibility options{/b}\nFollow the button instructions, then click elsewhere to progress\n{i}- Quit via the menu{/i}"
    $ stage += 1
    show screen DemoButton
    "{b}Example 2: Showing a non-modal screen, total cover{/b}\nFollow the button instructions, then click elsewhere to progress\n{i}- Quit via the menu{/i}"
    $ stage += 1
    show screen DemoButton
    "{b}Example 3: Showing a modal screen, partial cover{/b}\nFollow the button instructions, then click elsewhere to progress\n{i}- Quit via the menu{/i}"
    $ stage += 1
    show screen DemoButton
    "{b}Example 4: Showing a modal screen, no cover{/b}\nFollow the button instructions, then click elsewhere to progress\n{i}- Quit via the menu{/i}"
    $ stage += 1
    show screen DemoButton
    $ renpy.say("Conclusion:","'unhovered' is unable to either receive or act upon information about the focus change before 'modal True' prevents it from doing so. Nor can it recover after the fact.")
    $ stage = 0
    jump loop
```

**Comment by altskop** [comment]
This isn't a Ren'Py bug, but rather an issue in your own code.
```
screen DemoButton():
    default hasfocus = False
```
Declares a default variable that keeps its value. After the screen is redrawn, the value stays the same, as `default` keyword does not override existing variables.

What you want to do is declare a local variable instead, like this:
```
screen DemoButton():
    $ hasfocus = False
```
With this little change, in all the scenarios unhovered functions as intended, because the variable is set to False every time the screen is redrawn.

**Comment by JSearUK** [comment]
With your change in place, the text no longer changes color or has its transform activated at all, rendering whether or not
```py
        hovered SetScreenVariable("hasfocus", True)
        unhovered SetScreenVariable("hasfocus", False)
```
works or not rather moot.

I'm trying to manage the state of child elements based upon whether or not the parent has focus, such as buttons-within-buttons. It works just fine until *what I assume to be* `modal` interferes, but, since I'm just a hobbyist, I made this demo to try to illustrate not-working-as-expected behaviour: the button loses focus and, for whatever reason, doesn't change that variable.

**Comment by renpytom** [comment]
Can you explain what you think it should be doing, and what it's not doing? I appreciate the example, but it's hard to determine what's going wrong without knowing what you expect.

**Comment by JSearUK** [comment]
- The button, upon gaining focus, uses `hovered` to toggle a local variable named `hasfocus` (defaulted as False) to True via SetScreenVariable().
- `hasfocus` is used to conditionally apply properties to one or more children, in this case resulting in the text child of the button scrolling vertically and changing color. `hovered` has worked.
- Under almost all circumstances, `unhovered` will toggle `hasfocus` back to False; the text returns to its initial state. However, if the action of the button opens a `modal True` screen (or the Accessibility screen), neither the text color nor the transform are reverted if the mouse has moved off the button.
- This leads me to believe that, for some reason, `modal True` is preventing `unhovered` from doing its job: in this case, changing `hasfocus` back to False.
- I do not know what the cause of this discrepancy might actually be, despite my guesswork, so I brought it up here for investigation.

**Comment by renpytom** [comment]
I'll look into this and figure out what's going on.

**Comment by Andykl** [comment]
This is basically #2743

**Comment by renpytom** [comment]
If I understand the test correctly, this was fixed by 1ec2108354cdac5f2a1d3574c9bb2613463d32f9 , which also fixed #2743.
